### ISSUE TITLE
Can not decode payment response ,  TimeStamp is string, not date so decode fail.  

### ISSUE THREAD

**Issue by jonhcbs** [Adyen/adyen-terminal-api-ios#27]
**Describe the bug**
Can not decode payment response   TimeStamp is string, not date so decode fail.  any sugguestion?
  "TimeStamp": "2024-08-17T12:46:49.000Z",

**To Reproduce**
1. initialize a transaction using demo https://github.com/Adyen/adyen-pos-mobile-ios
2. make a payment and see the response, and decode the response to PaymentResponse object
3. decode fails

▿ DecodingError
  ▿ typeMismatch : 2 elements
    - .0 : Swift.Double
    ▿ .1 : Context
      ▿ codingPath : 3 elements
        - 0 : CodingKeys(stringValue: "SaleData", intValue: nil)
        - 1 : CodingKeys(stringValue: "SaleTransactionID", intValue: nil)
        - 2 : CodingKeys(stringValue: "TimeStamp", intValue: nil)
      - debugDescription : "Expected to decode Double but found a string instead."
      - underlyingError : nil

**Expected behavior**
decode success

**Screenshots**
  ▿ typeMismatch : 2 elements
    - .0 : Swift.Double
    ▿ .1 : Context
      ▿ codingPath : 3 elements
        - 0 : CodingKeys(stringValue: "SaleData", intValue: nil)
        - 1 : CodingKeys(stringValue: "SaleTransactionID", intValue: nil)
        - 2 : CodingKeys(stringValue: "TimeStamp", intValue: nil)
      - debugDescription : "Expected to decode Double but found a string instead."
      - underlyingError : nil
 
**Smartphone (please complete the following information):**
 - Device:  ipad 9
 - OS: 17.6

**Additional context**

`
{
    "SaleToPOIResponse": {
        "PaymentResponse": {
            "POIData": {
                "POITransactionID": {
                    "TimeStamp": "2024-08-17T12:46:49.000Z",
                    "TransactionID": "58m7001723898809000.BTRM64JZCX9QJ2V5"
                },
                "POIReconciliationID": "0"
            },
            "PaymentReceipt": [
                {
                    "DocumentQualifier": "CashierReceipt",
                    "OutputContent": {
                        "OutputFormat": "Text",
                        "OutputText": [
                            {
                                "EndOfLineFlag": true,
                                "CharacterStyle": "Bold",
                                "Text": "key=header1"
                            },
                            {
                                "CharacterStyle": "Bold",
                                "Text": "key=header2",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=MERCHANT%20COPY&key=merchantTitle",
                                "CharacterStyle": "Bold"
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Date&value=08%2f17%2f2024&key=txdate",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Time&value=07%3a46%3a49&key=txtime",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Card&value=%2a%2a%2a%2a0002&key=pan",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Pref.%20name&value=mc%20en%20usa%20usd&key=preferredName",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Card%20type&value=mcpremiumcredit&key=cardType",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Payment%20method&value=mc&key=paymentMethod",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Payment%20variant&value=mcpremiumcredit&key=paymentMethodVariant"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Entry%20mode&value=ICC&key=posEntryMode"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=AID&value=A000000004101002&key=aid"
                            },
                            {
                                "Text": "name=MID&value=50&key=mid",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=TID&value=B5FF9A7E-1BFD-4998-A147-33A2C7A9FFD4.360&key=tid"
                            },
                            {
                                "Text": "name=PTID&value=B5FF9A7E&key=ptid",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "Text": "name=Auth.%20code&value=123456&key=authCode",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Tender&value=58m7001723898809000&key=txRef",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Reference&value=614cab19-f608-409d-92b1-ac99bd25d193&key=mref",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "Text": "name=Type&value=GOODS_SERVICES&key=txtype",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "CharacterStyle": "Bold",
                                "Text": "name=TOTAL&value=%24%2025.14&key=totalAmount"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=APPROVED&key=approved",
                                "CharacterStyle": "Bold"
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            }
                        ]
                    },
                    "RequiredSignatureFlag": false
                },
                {
                    "RequiredSignatureFlag": false,
                    "OutputContent": {
                        "OutputFormat": "Text",
                        "OutputText": [
                            {
                                "EndOfLineFlag": true,
                                "CharacterStyle": "Bold",
                                "Text": "key=header1"
                            },
                            {
                                "Text": "key=header2",
                                "CharacterStyle": "Bold",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=CARDHOLDER%20COPY&key=cardholderHeader",
                                "CharacterStyle": "Bold"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "Text": "name=Date&value=08%2f17%2f2024&key=txdate",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Time&value=07%3a46%3a49&key=txtime"
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=Card&value=%2a%2a%2a%2a0002&key=pan",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Pref.%20name&value=mc%20en%20usa%20usd&key=preferredName"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Card%20type&value=mcpremiumcredit&key=cardType"
                            },
                            {
                                "Text": "name=Payment%20method&value=mc&key=paymentMethod",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Payment%20variant&value=mcpremiumcredit&key=paymentMethodVariant"
                            },
                            {
                                "Text": "name=Entry%20mode&value=ICC&key=posEntryMode",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=AID&value=A000000004101002&key=aid"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=MID&value=50&key=mid"
                            },
                            {
                                "Text": "name=TID&value=B5FF9A7E-1BFD-4998-A147-33A2C7A9FFD4.360&key=tid",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=PTID&value=B5FF9A7E&key=ptid",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Auth.%20code&value=123456&key=authCode"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Tender&value=58m7001723898809000&key=txRef"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Reference&value=614cab19-f608-409d-92b1-ac99bd25d193&key=mref"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "Text": "name=Type&value=GOODS_SERVICES&key=txtype",
                                "EndOfLineFlag": true
                            },
                            {
                                "Text": "name=TOTAL&value=%24%2025.14&key=totalAmount",
                                "EndOfLineFlag": true,
                                "CharacterStyle": "Bold"
                            },
                            {
                                "Text": "key=filler",
                                "EndOfLineFlag": true
                            },
                            {
                                "CharacterStyle": "Bold",
                                "Text": "name=APPROVED&key=approved",
                                "EndOfLineFlag": true
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "key=filler"
                            },
                            {
                                "EndOfLineFlag": true,
                                "Text": "name=Retain%20for%20your%20records&key=retain"
                            },
                            {
                                "Text": "name=Thank%20you&key=thanks",
                                "EndOfLineFlag": true
                            }
                        ]
                    },
                    "DocumentQualifier": "CustomerReceipt"
                }
            ],
            "SaleData": {
                "SaleTransactionID": {
                    "TransactionID": "614cab19-f608-409d-92b1-ac99bd25d193",
                    "TimeStamp": "2024-08-17T12:46:44.810Z"
                }
            },
            "PaymentResult": {
                "AmountsResp": {
                    "Currency": "USD",
                    "AuthorizedAmount": 25.14
                },
                "PaymentAcquirerData": {
                    "ApprovalCode": "123456",
                    "AcquirerTransactionID": {
                        "TransactionID": "BTRM64JZCX9QJ2V5",
                        "TimeStamp": "2024-08-17T12:46:49.000Z"
                    },
                    "MerchantID": "ChowbusUS",
                    "AcquirerPOIID": "B5FF9A7E-1BFD-4998-A147-33A2C7A9FFD4.360"
                },
                "OnlineFlag": true,
                "PaymentInstrumentData": {
                    "CardData": {
                        "MaskedPan": "510006 **** 0002",
                        "EntryMode": [
                            "ICC"
                        ],
                        "PaymentBrand": "mc",
                        "SensitiveCardData": {
                            "ExpiryDate": "1229"
                        }
                    },
                    "PaymentInstrumentType": "Card"
                }
            },
            "Response": {
                "AdditionalResponse": "AID=A000000004101002&acquirerResponseCode=APPROVED&alias=K922116043233497&aliasType=Default&applicationLabel=MCENUSAUSD&applicationPreferredName=mc%20en%20usa%20usd&authCode=123456&avsResult=0%20Unknown&cardBin=510006&cardHolderVerificationMethodResults=1F0002&cardScheme=mcpremiumcredit&cardSummary=0002&cardType=mc&cvcResult=0%20Unknown&expiryMonth=12&expiryYear=2029&giftcardIndicator=false&iso8601TxDate=2024-08-17T12%3a46%3a49.000Z&merchantReference=614cab19-f608-409d-92b1-ac99bd25d193&mid=50&offline=false&paymentMethod=mc&paymentMethodVariant=mcpremiumcredit&posAmountCashbackValue=0&posAmountGratuityValue=0&posAuthAmountCurrency=USD&posAuthAmountValue=2514&posEntryMode=ICC&posOriginalAmountValue=2514&posadditionalamounts.originalAmountCurrency=USD&posadditionalamounts.originalAmountValue=2514&pspReference=BTRM64JZCX9QJ2V5&shopperCountry=US&store=r10538&tc=9E6A4AAE80858100&tid=B5FF9A7E&transactionReferenceNumber=BTRM64JZCX9QJ2V5&transactionType=GOODS_SERVICES&txdate=17-08-2024&txtime=07%3a46%3a49",
                "Result": "Success"
            }
        },
        "MessageHeader": {
            "MessageClass": "Service",
            "MessageType": "Response",
            "ProtocolVersion": "3.0",
            "SaleID": "76C7E298-0AD7-4EE2-8389-161466A8DD06",
            "ServiceID": "4FA02A",
            "POIID": "B5FF9A7E-1BFD-4998-A147-33A2C7A9FFD4.360",
            "MessageCategory": "Payment"
        }
    }
}
`

**Comment by kaphacius** [comment]
hey @jonhcbs ,

thanks for opening the issue.

for decoding the response, could you try using `Coder.decode` function from the library? since the date format is very specific we provide our own `JSONEncoder`/`JSONDecoder` implementations.
```swift
let response = try Coder.decode(Message<PaymentResponse>.self, from: responseData)
```

Cheers

**Comment by jonhcbs** [comment]
it works, great.

**Comment by jonhcbs** [comment]
https://github.com/Adyen/adyen-pos-mobile-ios
https://github.com/Adyen/adyen-pos-mobile-android

btw: could help also open issues of pos repo,  so that could also rise one if have problem. thanks.

**Comment by kaphacius** [comment]
hey @jonhcbs . unsure what you mean, sorry.
can you rephrase please?
